File: backend-go/internal/service/normalize.go
package service

import (
	"regexp"
	"strings"
)

var whitespaceRe = regexp.MustCompile(`\s+`)

func Normalize(text string, ignoreWhitespace bool) string {
	if text == "" {
		return ""
	}

	result := strings.TrimSpace(text)
	if ignoreWhitespace {
		result = whitespaceRe.ReplaceAllString(result, "")
	}

	result = strings.NewReplacer(
		"，", ",",
		"。", ".",
		"；", ";",
		"（", "(",
		"）", ")",
		"：", ":",
	).Replace(result)
	return result
}

File: frontend/src/lib/compare/normalize.ts
const WHITESPACE = /\s+/g

export function normalize(text: string, ignoreWhitespace: boolean): string {
  if (!text) return ''

  let result = text.trim()
  if (ignoreWhitespace) {
    result = result.replace(WHITESPACE, '')
  }

  return result
    .replace(/，/g, ',')
    .replace(/。/g, '.')
    .replace(/；/g, ';')
    .replace(/（/g, '(')
    .replace(/）/g, ')')
    .replace(/：/g, ':')
}
